src/controllers/key.controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Key } from "../models/key.model";
+import { keyStore } from "../store";
+import { ACTIONS } from "../types";
+
+export class KeyController<T extends HTMLElement> {
+  private readonly key: Key;
+  private readonly audio: HTMLAudioElement;
+  private readonly element: T;
+  private prevClickCount: number = 0;
```

Comment:
Controller should not require to maintain any data like this. This is purely a model's responsibility.